libminifi/src/FlowController.cpp

```diff
@@ -127,35 +127,34 @@ bool FlowController::applyConfiguration(const std::string &source, const std::st
 
   logger_->log_info("Starting to reload Flow Controller with flow control name %s, version %d", newRoot->getName(), newRoot->getVersion());
 
-  updating_.beginUpdate();
   bool started = false;
-
   {
-    std::lock_guard<std::recursive_mutex> flow_lock(mutex_);
-    stop();
-
-    root_wrapper_.setNewRoot(std::move(newRoot));
-    initialized_ = false;
-    try {
-      load(true);
-      started = start() == 0;
-    } catch (const std::exception& ex) {
-      logger_->log_error("Caught exception while starting flow, type %s, what: %s", typeid(ex).name(), ex.what());
-    } catch (...) {
-      logger_->log_error("Caught unknown exception while starting flow, type %s", getCurrentExceptionTypeName());
-    }
-    if (!started) {
-      logger_->log_error("Failed to start new flow, restarting previous flow");
-      root_wrapper_.restoreBackup();
-      load(true);
-      start();
-    } else {
-      root_wrapper_.clearBackup();
+    auto update_lock = updating_.getUpdateLock();
+    {
+      std::lock_guard<std::recursive_mutex> flow_lock(mutex_);
```

Comment:
The inner scope is not necessary here, the two locks could just follow each other.

Reply:
Updated in 1fd1685fbccb60386133baeee87ddcffa4747487